src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
> In order to serve to get mode clients directly via IP (i.e., no domain fronting), flashlight servers would need to listen through HTTPS.

That's not the only thing that would need to change, since domain-fronting uses the enproxy protocol, which is unnecessary for chained proxying.  Flashlight (now Lantern) can be modified to support the use of chained proxying using the chained package (which already has a server component).  It would also need to implement the Apache HTTP server mimicry that the fallbacks currently do, along with the auth_token checking.